plugins/filter/to_prettytable.py

```diff
@@ -0,0 +1,380 @@
+    """Find a matching key in a dictionary, handling type conversion.
+
+    Args:
+        item_dict: Dictionary to search in
+        lookup_key: Key to look for, possibly needing type conversion
+
+    Returns:
+        The matching key or None if no match found
+    """
+    # Direct key match
+    if lookup_key in item_dict:
+        return lookup_key
+
+    # Try boolean conversion for 'true'/'false' strings
+    if isinstance(lookup_key, string_types):
+        if lookup_key.lower() == 'true' and True in item_dict:
+            return True
+        if lookup_key.lower() == 'false' and False in item_dict:
+            return False
+
+        # Try numeric conversion for string numbers
+        if lookup_key.isdigit() and int(lookup_key) in item_dict:
+            return int(lookup_key)
+
+    # No match found
+    return None
+
+
+def _build_key_maps(data):
+    """Build mappings between string keys and original keys.
+
+    Args:
+        data: Dictionary with keys to map
+
+    Returns:
+        Tuple of (key_map, reverse_key_map)
+    """
+    key_map = {}
+    reverse_key_map = {}
+
+    first_dict = data[0]
+    for orig_key in first_dict.keys():
+        # Store string version of the key
+        str_key = to_text(orig_key)
+        key_map[str_key] = orig_key
+        # Also store lowercase version for case-insensitive lookups
+        reverse_key_map[str_key.lower()] = orig_key
+
+    return key_map, reverse_key_map
+
+
+def _configure_alignments(table, field_names, column_alignments):
+    """Configure column alignments for the table.
+
+    Args:
+        table: The PrettyTable instance to configure
+        field_names: List of field names to align
+        column_alignments: Dict of column alignments
+    """
+    valid_alignments = {"left", "center", "right", "l", "c", "r"}
+
+    if not isinstance(column_alignments, dict):
+        return
+
+    for col_name, alignment in column_alignments.items():
+        if col_name in field_names:
+            # We already validated alignment is a string and a valid value in the main function
+            # Just apply it here
+            alignment = alignment.lower()
+            table.align[col_name] = alignment[0]
+
+
+def to_prettytable(data, *args, **kwargs):
+    """Convert a list of dictionaries to an ASCII table.
+
+    Args:
+        data: List of dictionaries to format
+        *args: Optional list of column names to specify column order
+        **kwargs: Optional keyword arguments:
+            - column_order: List of column names to specify the order
+            - header_names: List of custom header names
+            - column_alignments: Dict of column alignments (left, center, right)
+
+    Returns:
+        String containing the ASCII table
+    """
+    if not HAS_PRETTYTABLE:
+        raise AnsibleFilterError(
+            'You need to install "prettytable" Python module to use this filter'
+        )
+
+    # Handle empty data
+    if not data:
+        return "++\n++"
```

Comment:
I'd move this below the validation of the options.

Reply:
I tested such approach. If we process empty data after other options and data processing, it causes exception: `IndexError: list index out of range`

Reply:
Some parameter validations depend on the input data type. We validate the size of certain parameters, such as `column_order` and `header_names`, by comparing them to the number of fields in the input dictionary. If we process empty data after that, it may lead to an exception like the one described above.

Reply:
Now this can be moved down after the argument validation, I think.

Reply:
If we change the code this way, it will make it impossible to use the filter with custom header parameters while still allowing dynamic processing of input data (including zero-length lists). As a result, if we receive an empty list as input (which can happen in some cases) and `header_names` and `column_order` parameters are set, we would always get an error instead of a zero-length table. I believe this would make the filter less convenient to use.

Reply:
Then please adjust these specific validations accordingly. Simply not validating any other parameter **at all** when `data` is empty is wrong.

Reply:
Adjusted the code to handle validations properly even with empty input data. Added integration tests for new use cases with empty data.